request_listener/curl.go

```diff
@@ -0,0 +1,67 @@
+package request_listener
+
+import (
+	"bytes"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"strings"
+)
+
+func CurlLogDumper(request *http.Request) {
+	if request == nil {
+		return
+	}
+	cmd, err := GetCurlCommand(request)
+	if err != nil {
+		log.Println("cannot print curl command", err)
+		return
+	}
+	log.Println(cmd)
+}
+
+func GetCurlCommand(req *http.Request) (*Cmd, error) {
+	cmd := &Cmd{
+		"curl",
+		"-X", escape(req.Method),
+		escape(req.URL.String()),
+	}
+
+	if req.Body != nil {
+		body, err := ioutil.ReadAll(req.Body)
+		if err != nil {
+			return nil, err
+		}
+		req.Body = nopCloser{bytes.NewBuffer(body)}
+		if len(string(body)) > 0 {
+			cmd.append("-d", escape(string(body)))
+		}
+	}
+
+	for h := range req.Header {
```

Comment:
Maybe handle cookies in another iteratio ? (there are so much options to handle 😨)